torch/testing/_asserts.py

```diff
@@ -154,49 +154,37 @@ def _equalize_attributes(a: torch.Tensor, b: torch.Tensor) -> Tuple[torch.Tensor
     return a, b
 
 
-_Trace = namedtuple("Trace", ("total", "abs", "rel", "idx", "diff", "a", "b"))
-
-
-def _trace_mismatches(a: torch.Tensor, b: torch.Tensor, mismatches: torch.Tensor) -> _Trace:
-    """Traces mismatches and returns the found information.
-
-    The returned named tuple has the following fields:
-    - total (int): Total number of values in :attr:`a` and :attr:`b`.
-    - abs (int): Absolute number of mismatches.
-    - rel (float): Relative number of mismatches.
-    - idx (Union[int, Tuple[int, ...]]): Index of greatest absolute difference.
-    - diff (Union[int, float]): Greatest absolute difference.
-    - a (Union[int, float]): Value of :attr:`a` at the greatest absolute difference.
-    - b (Union[int, float]): Value of :attr:`a` at the greatest absolute difference.
-
-    For ``diff``, ``a``, and ``b`` the returned type depends on the :attr:`~torch.Tensor.dtype` of :attr:`a` and
-    :attr:`b`.
+def _trace_mismatches(a: torch.Tensor, b: torch.Tensor, mismatches: torch.Tensor) -> str:
+    """Traces mismatches.
 
     Args:
         a (torch.Tensor): First tensor.
         b (torch.Tensor): Second tensor.
         mismatches (torch.Tensor): Boolean mask of the same shape as :attr:`a` and :attr:`b` that indicates the
             location of mismatches.
+
+    Returns:
+        (str): Multiline list comprising the absolute and relative number of mismatches, and the maximum absolute and
+            relative difference with the corresponding index.
     """
-    total = mismatches.numel()
-    abs = torch.sum(mismatches).item()
-    rel = abs / total
+    total_elements = mismatches.numel()
+    abs_mismatches = torch.sum(mismatches).item()
+    rel_mismatches = abs_mismatches / total_elements
 
     dtype = torch.float64 if a.dtype.is_floating_point else torch.int64
     a_flat = a.flatten().to(dtype)
     b_flat = b.flatten().to(dtype)
 
-    abs_diff_flat = torch.abs(a_flat - b_flat)
-    idx_flat = torch.argmax(abs_diff_flat)
-
-    return _Trace(
-        total=total,
-        abs=abs,
-        rel=rel,
-        idx=_unravel_index(idx_flat, a.shape),
-        diff=abs_diff_flat[idx_flat].item(),
-        a=a_flat[idx_flat].item(),
-        b=b_flat[idx_flat].item(),
+    abs_diff = torch.abs(a_flat - b_flat)
+    max_abs_diff, max_abs_diff_flat_idx = torch.max(abs_diff, 0)
+
+    rel_diff = abs_diff / torch.abs(b_flat)
+    max_rel_diff, max_rel_diff_flat_idx = torch.max(rel_diff, 0)
+
+    return (
+        f"Mismatched elements: {abs_mismatches} / {total_elements} ({rel_mismatches:.1%})\n"
+        f"Max. abs. diff.: {max_abs_diff.item()} at {_unravel_index(max_abs_diff_flat_idx.item(), mismatches.shape)}\n"
```

Comment:
It might be interesting to return these statistics as data. Then the close assertion could create its own string that inserted the greatest absolute difference and greatest relative difference that would be allowed.

Reply:
So something like https://github.com/pytorch/pytorch/pull/55145#issuecomment-817297273?